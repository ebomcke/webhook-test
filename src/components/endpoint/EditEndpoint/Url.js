import React from 'react';
import { compose } from 'recompose';
import { withAccount } from '../../../contexts/Account';
import { Form, Input } from 'semantic-ui-react';

const Url = ({ endpointEditor, account, nextButton, disabled }) => (
  <Form>
    <Form.Field required>
      <label>URL</label>
      <Input
        disabled={disabled}
        label={`${process.env.API_URL}/webhook/${account.organisationName}/`}
        placeholder="my/custom/url"
        name="path"
        value={endpointEditor.endpoint.path}
        onChange={(event, data) => endpointEditor.onUrlChange(data.value)}
      />
    </Form.Field>
    {!!nextButton && nextButton()}
  </Form>
);

export default compose(withAccount)(Url);
