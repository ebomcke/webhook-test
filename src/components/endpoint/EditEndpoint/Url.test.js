import React from 'react';
import { create } from 'react-test-renderer';
import Url from './Url';

const endpointEditor = {
  onUrlChange: jest.fn(),
  endpoint: {
    path: 'unit-test-path',
  },
};

const nextButton = () => <div>Next button placeholder</div>;

describe('Rendering tests', () => {
  test('Without next button', () => {
    const url = create(
      <Url
        endpointEditor={endpointEditor}
        nextButton={false}
      />,
    );
    expect(url.toJSON()).toMatchSnapshot();
  });

  test('With next button', () => {
    const url = create(
      <Url
        endpointEditor={endpointEditor}
        nextButton={nextButton}
      />,
    );
    expect(url.toJSON()).toMatchSnapshot();
  });

  test('With url input disabled', () => {
    const url = create(
      <Url
        endpointEditor={endpointEditor}
        nextButton={nextButton}
        disabled={true}
      />,
    );
    expect(url.toJSON()).toMatchSnapshot();
  });
});

describe('Interactions tests', () => {
  const url = create(
    <Url
      endpointEditor={endpointEditor}
      nextButton={nextButton}
      disabled={true}
    />,
  );
  
  test('Changing the url updates the endpoint editor', () => {
    const urlInput = url.root.findByProps({ name: 'path' });
    const newPath = 'unit-test-updated-path';
    urlInput.props.onChange(null, { value: newPath });
    expect(endpointEditor.onUrlChange.mock.calls.length).toBe(1);
    expect(endpointEditor.onUrlChange.mock.calls[0][0]).toEqual(
      newPath,
    );
  });
});
