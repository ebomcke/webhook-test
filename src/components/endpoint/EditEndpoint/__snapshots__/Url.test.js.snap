// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering tests With next button 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="required field"
  >
    <label>
      URL
    </label>
    <div
      className="ui labeled input"
    >
      <div
        className="ui label label"
        onClick={[Function]}
      >
        https://webhook-tester-1547141127436.appspot.com/webhook/unit-test-org/
      </div>
      <input
        name="path"
        onChange={[Function]}
        placeholder="my/custom/url"
        type="text"
        value="unit-test-path"
      />
    </div>
  </div>
  <div>
    Next button placeholder
  </div>
</form>
`;

exports[`Rendering tests With url input disabled 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="required field"
  >
    <label>
      URL
    </label>
    <div
      className="ui disabled labeled input"
    >
      <div
        className="ui label label"
        onClick={[Function]}
      >
        https://webhook-tester-1547141127436.appspot.com/webhook/unit-test-org/
      </div>
      <input
        disabled={true}
        name="path"
        onChange={[Function]}
        placeholder="my/custom/url"
        tabIndex={-1}
        type="text"
        value="unit-test-path"
      />
    </div>
  </div>
  <div>
    Next button placeholder
  </div>
</form>
`;

exports[`Rendering tests Without next button 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="required field"
  >
    <label>
      URL
    </label>
    <div
      className="ui labeled input"
    >
      <div
        className="ui label label"
        onClick={[Function]}
      >
        https://webhook-tester-1547141127436.appspot.com/webhook/unit-test-org/
      </div>
      <input
        name="path"
        onChange={[Function]}
        placeholder="my/custom/url"
        type="text"
        value="unit-test-path"
      />
    </div>
  </div>
</form>
`;
